src/veins/base/toolbox/Signal.cc

```diff
@@ -555,268 +555,172 @@ Signal& Signal::operator+=(const Signal& other)
 {
     assert(this->getSpectrum() == other.getSpectrum());
 
+#ifndef PREFER_VECTORIZATION
     size_t lowIndex = std::min(this->getRelativeStart(), other.getRelativeStart());
     size_t highIndex = std::max(this->getRelativeEnd(), other.getRelativeEnd());
+#else
+    size_t lowIndex = 0;
+    size_t highIndex = numAbsoluteValues;
+#endif
 
-    for (size_t i = lowIndex; i < highIndex; i++) {
-        (*this)[i] += other[i];
-    }
+    std::transform(values + lowIndex, values + highIndex, other.values + lowIndex, values + lowIndex, std::plus<double>());
     return *this;
 }
 
 Signal& Signal::operator+=(const double value)
 {
-    for (size_t i = 0; i < numAbsoluteValues; i++) {
-        (*this)[i] += value;
-    }
+    std::transform(values, values + numAbsoluteValues, values, [value](double other) { return other + value; });
     return *this;
 }
 
 Signal& Signal::operator-=(const Signal& other)
 {
     assert(this->getSpectrum() == other.getSpectrum());
 
+#ifndef PREFER_VECTORIZATION
```

Comment:
Suggestion: Isolate in an (inline) function.
And to ease reading, I would use `ifdef` instead of `ifndef`. Maybe even add a comment like
```
// covering whole spectrum when using vectorization to preserve alignment.
```
and 
```
// only covering actually set portion of the spectrum
```